implants/eldritch/src/pivot/ncat_impl.rs

```diff
@@ -0,0 +1,172 @@
+
+use std::net::Ipv4Addr;
+
+use anyhow::Result;
+use tokio::io::{AsyncWriteExt, AsyncReadExt};
+use tokio::net::{TcpStream, UdpSocket};
+
+// Since we cannot go from async (test) -> sync (ncat) `block_on` -> async (handle_ncat) without getting an error "cannot create runtime in current runtime since current thread is calling async code."
+async fn handle_ncat(address: String, port: i32, data: String, protocol: String) -> Result<String> {
```

Comment:
The working portion and the primary function have been separated to make testing async code easier.